docs/source/ProgrammingGuide/examples/graph_capture.cpp

```diff
@@ -0,0 +1,123 @@
+#include "cublas_v2.h"
+
+#include "Kokkos_Core.hpp"
+#include "Kokkos_Graph.hpp"
+
+#define CHECK_CUBLAS_CALL(call)                           \
+  {                                                       \
+    const auto error_code = call;                         \
+    if(error_code != CUBLAS_STATUS_SUCCESS)               \
+    {                                                     \
+      printf("%s:%d: failure of statement %s: %s (%d)\n", \
+        __FILE__, __LINE__,                               \
+        #call,                                            \
+        cublasGetStatusName(error_code), error_code);     \
+      std::abort();                                       \
+    }                                                     \
+  }
+
+template <typename MatrixType, typename VectorType>
+struct InitializeData
+{
+  MatrixType matrix;
+  VectorType vector;
+
+  template <typename T>
+  KOKKOS_FUNCTION
+  void operator()(const T irow) const
+  {
+      matrix(2 * irow)     = 2 * irow + 1;
+      matrix(2 * irow + 1) = 2 * irow + 2;
+
+      vector(irow) = irow + 5;
+  }
+};
+
+auto create_cublas_handle()
+{
+  cublasHandle_t handle = nullptr;
+
+  CHECK_CUBLAS_CALL(cublasCreate(&handle));
+
+  return std::shared_ptr<cublasContext>(handle, [](cublasHandle_t ptr) {
+    CHECK_CUBLAS_CALL(cublasDestroy(ptr));
+  });
+}
+
+template <typename Exec, typename Predecessor, typename MatrixType, typename VectorType, typename ResultType>
+auto gemv(const Exec& exec, const Predecessor& predecessor, const MatrixType& matrix, const VectorType& vector, const ResultType& result)
+{
+  static_assert(std::is_same_v<typename MatrixType::value_type, double>);
+  static_assert(std::is_same_v<typename VectorType::value_type, double>);
+  static_assert(std::is_same_v<typename ResultType::value_type, double>);
+
+  auto handle = create_cublas_handle();
+
+  const double alpha = 1., beta = 1.;
+
+  /// The @c handle is a shared resource stored in the lambda.
+  /// Since the lambda is stored by the node, and the node won't be destroyed
+  /// until the @c Kokkos::Graph is destroyed, the @c cuBLAS handle is
+  /// guaranteed to stay alive for the whole graph lifetime.
+  return predecessor.cuda_capture(
+    exec,
+    [handle = std::move(handle), matrix, vector, result, alpha, beta](const Kokkos::Cuda& exec_) {
```

Comment:
you might want to explain why the handle should be moved here ... moving a `shared_ptr` smells a bit

Reply:
I'm doing it out of habit since in such circumstances copying the shared pointer would be a waste of CPU cycles. Probably compilers are already smart enough to do it under the hood already.

Reply:
```suggestion
    [=](const Kokkos::Cuda& exec_) {
```

Reply:
Indeed, let's use `=`.